core/src/main/java/jenkins/security/ResourceDomainConfiguration.java

```diff
@@ -134,7 +134,7 @@ private FormValidation checkUrl(String resourceRootUrlString, boolean allowOnlin
 
         StaplerRequest currentRequest = Stapler.getCurrentRequest();
         if (currentRequest != null) {
-            String currentRequestHost = currentRequest.getHeader("Host");
+            String currentRequestHost = String.format("%s:%s", currentRequest.getServerName(), currentRequest.getServerPort());
```

Comment:
Did you test this? Seems like the next condition would rarely be `true`, given the lack of a port for the majority of host specifications.

Reply:
I see, I was miss led by the original code, the getHeader("Host") could return host:port but the resourceRootUrlHost will always just be the host. I will made the change on the last commit

Reply:
Yeah you also kinda discovered a bug -- resource root URL validation doesn't look like it deals well with nonstandard parts.

Are you up to attempting to address this while you're here? 😀

Reply:
@daniel-beck would it be OK if this were limited to resolving the [reported null pointer exception](https://issues.jenkins.io/browse/JENKINS-70630)?

We've had two different bug reports of the issue and I'd rather see us bring that to a conclusion rather than broadening the pull request to cover more topics.

Reply:
Given that the requirement is that the hostname of the resource root host is different from the jenkins host, does the port check really matter? This check is to prevent the user from locking himself out of Jenkins by using the same hostname, I don't think the port would matter in this case. 
In my last commit, is now only checking that the current server hostname is different from what the user is trying to set as the resource root url.

Reply:
@ca-monteiro Good point, might be why I didn't initially bother to. So if this works it should be fine.

@MarkEWaite Hence me asking nicely and not marking the PR as "change requested" 😉

Reply:
@daniel-beck, yes correct, the Host header sometimes could have the port appended to the string. The **resourceRootUrlHost** string would never have the port as it comes from the **URL().getHost()**.